cirq-core/cirq/ops/state_preparation_channel.py

```diff
@@ -108,13 +111,24 @@ def _kraus_(self):
         return operator
 
     def __repr__(self) -> str:
-        return f'cirq.StatePreparationChannel({proper_repr(self._state)})'
+        return (
+            f'cirq.StatePreparationChannel('
+            f'target_state={proper_repr(self.state)}, name="{self._name}")'
+        )
+
+    def __str__(self) -> str:
+        return f'StatePreparationChannel({self.state})'
+
+    def _approx_eq_(self, other: Any, atol) -> bool:
+        if not isinstance(other, StatePreparationChannel):
+            return False
+        return np.allclose(self.state, other.state, rtol=0, atol=atol)
 
     def __eq__(self, other) -> bool:
         if not isinstance(other, StatePreparationChannel):
             return False
-        return np.allclose(self.state, other.state)
+        return np.array_equal(self.state, other.state)
```

Comment:
Should we also include the `name` parameter when checking for equality? 

If not, let's also add another test in `test_equality_of_gates` and `test_approx_equality_of_gates ` which tests that objects with different names but same states are equal.

Reply:
I checked `MatrixGate` and others, they all seem to not care about name of the gate when checking for equality, I added that to the tests.